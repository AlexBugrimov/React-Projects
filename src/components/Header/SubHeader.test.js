import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SubHeader from './SubHeader'

describe('the <SubHeader /> component', () => {
  const renderer = new ShallowRenderer();

  it('should render', () => {
    renderer.render(<SubHeader />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  })

  it('should render with a dynamic title', () => {
    renderer.render(<SubHeader title="Test Application" />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  })

  it('should render with a goback button', () => {
    renderer.render(<SubHeader goBack={() => {}} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  })

  it('should render with a form button', () => {
    renderer.render(<SubHeader openForm={() => {}} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  })
})
