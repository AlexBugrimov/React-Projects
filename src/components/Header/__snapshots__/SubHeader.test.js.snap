// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <SubHeader /> component should render 1`] = `
<ForwardRef>
  <ForwardRef />
</ForwardRef>
`;

exports[`the <SubHeader /> component should render with a dynamic title 1`] = `
<ForwardRef>
  <ForwardRef>
    Test Application
  </ForwardRef>
</ForwardRef>
`;

exports[`the <SubHeader /> component should render with a form button 1`] = `
<ForwardRef>
  <ForwardRef />
  <ForwardRef
    onClick={[Function]}
  >
    + Add Review
  </ForwardRef>
</ForwardRef>
`;

exports[`the <SubHeader /> component should render with a goback button 1`] = `
<ForwardRef>
  <ForwardRef
    onClick={[Function]}
  >
    &lt; Go Back
  </ForwardRef>
  <ForwardRef />
</ForwardRef>
`;
